### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by enoqueJonas**
# Blog App - processing data in models

## This PR was raised to:
- Set up associations between models.
- Write SQL queries with Active Record.

## What does this PR do:
- Create model classes 
- Set up associations between models

## Improvements to be made:
- Add validations

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :user
+
+  def post_counter
+    user.update(post_counter: user.posts.count)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/post.rb]: - [ ] Kindly I recommend you to  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the **#post_counter** method every time a new post has been saved to database, then make the **post_counter** method private Just like below.

```
  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

after_save :post_counter

  private

  def post_counter
        user.update(post_counter: user.posts.count)
  end

```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/comment.rb:11]: - [ ] Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the **#update_comments_counter** method every time a new comment has been saved to database, then make the **update_comments_counter** method private Just like below.

```
after_save :update_comments_counter

  private

  def update_comments_counter
           post.update(comments_counter: post.comments.count)
  end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/like.rb:11]: - [ ] Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the **#update_likes_counter** method every time a new like has been saved to database, then make the **update_likes_counter** method private Just like below.

```
after_save :update_likes_counter

  private

  def update_likes_counter
    post.update(likes_counter: post.likes.count)
  end
```

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @enoqueJonas   , 

Great job for implemented  the Project requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻

✔️ No Linters error
✔️ You used correct Git -Flow.
✔️ Good commit messages
✔️ Professional READEME 


almost

## Required Changes ♻️
_Check the comments below the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @enoqueJonas , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

##  Highlights 
:+1: no linter errors 
:+1: files structure

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**